### DISCUSSION TITLE
Introducing a secure hashing algorithm in the attributesprocessor

### DISCUSSION THREAD

**Discussion by florianrudaj** [open-telemetry/opentelemetry-collector-contrib#5318] [Ideas]
My organization wants to obscure the content of some attributes in spans for privacy reasons. However, the attributes that contain sensitive data should not be deleted. They want to be able to recognize a correlation between certain values and an error case despite the obscured data.   

At the moment it is possible to hash certain attributes by the attributesprocessor. SHA-1 is used as hashing method here even though it is insecure. However, to ensure privacy, my organization would like to use a secure hashing method. I would like to discuss if it makes sense to offer a secure hashing method besides SHA-1 or even to replace SHA-1 completely with it. I also don't understand the rationale behind favoring SHA-1 over secure hashing methods.

If it comes to the decision to use a secure hashing method I would like to implement this new functionality.

**Comment by atoulme** [comment]
See https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/19779 - does it do the job for you?

**Reply by florianrudaj** [reply]
Looks good
